apps/aot/src/app/leaderboard/_components/overall-leaderboard.tsx

```diff
@@ -33,15 +41,37 @@ async function getOverallLeaderboard(currentAdventDay: number) {
     ) r ON u.id = r.userId
   GROUP BY r.userId, u.name, u.image
   ORDER BY totalPoints DESC
-  LIMIT ${LEADERBOARD_RANKING_LIMIT};
-`;
+  LIMIT ${LEADERBOARD_RANKING_LIMIT};`;
+
+  // Prisma doesn't suppport distinct for .count()...
+  const challengeIdToday = ADVENT_CHALLENGE_IDS[currentAdventDay - 1];
+  const numberOfSubmissionsTodayPromise = prisma.$queryRaw<[{ count: number }]>`
+    SELECT COUNT(DISTINCT userId) as count
+    FROM Submission
+    WHERE challengeId = ${challengeIdToday}
+    AND isSuccessful = 1`;
+
+  const [ranking, [{ count: numberOfSubmissionsToday }]] = await Promise.all([
+    rankingPromise,
+    numberOfSubmissionsTodayPromise,
+  ]);
+
+  // Once we have top 100 for today, we can cache until midnight (next challenge release)
+  if (Number(numberOfSubmissionsToday) >= LEADERBOARD_RANKING_LIMIT) {
+    await redisClient.set(
+      'aot-overall-leaderboard',
+      JSON.stringify(ranking, (_, value) => (typeof value === 'bigint' ? value.toString() : value)),
+      { PXAT: getNextAdventDay() },
+    );
```

Comment:
`getNextAdventDay()` returns .getTime() of next release date and time.

`PXAT` will automatically expire the cache key we set here at that time.